calculator.js

```diff
@@ -157,18 +156,20 @@ function handlesEquals(){
     num1 = parseToNum(num1);
     num2 = parseToNum(num2); 
     result = calculateResult(operator, num1, num2);
-    num1 = num2 = operator = undefined;
-    operatorState = false; 
-    btnDecSeparator.classList.remove('dot-disabled');
+    doAfterGettingResult(result, 'equals');
 }
 
 function handlesDecSeparator(value){
-    showOnScreen(value);
+    showOnScreen(value,'add');
     btnDecSeparator.className += ' dot-disabled';
 }
 
-function showOnScreen(value){
-    digitsOnScreen.innerHTML += value;
+function showOnScreen(value, operation='replace'){
```

Comment:
It's preferable to build 2 distinct function to maintain it simple.
- putOnScreen - can be your replace operation
- appendOnScreen - can be your add operation

Other thing is the operator is a string, so if we want to change in the future this labels `add` and `replace` the developer responsible for this change will need to chase all the places in the code to change, open opportunities to mistakes.

Using functions we can use some editor to replace the function name.
Functions with a single responsibility reveal better the intention that they are doing.

Reply:
ok, broke the function into replaceOnScreen() and addToScreen()